orientdb/src/main/java/com/yahoo/ycsb/db/OrientDBClient.java

```diff
@@ -115,6 +116,9 @@ public void init() throws DBException {
 
   @Override
   public void cleanup() throws DBException {
+    // Set this thread's db reference (needed for thread safety in testing)
+    ODatabaseRecordThreadLocal.INSTANCE.set(db);
```

Comment:
Yes, this was a weird. I was not able to nail down exactly what is happening. When you run the tests, I am creating a OrientDBClient instance by hand and calling its ```init()```, ```cleanup()```, and operational methods manually. Two threads are used, what I don't understand is exactly what is happening on each. I can say that ```cleanup()``` seems to be running on a different thread then ```init()```, using ```ODatabaseRecordThreadLocal.INSTANCE.set(db);``` in ```init()``` did not fix the problem. Only from ```cleanup()``` did it fix the problem.